model/inline/src/main/scala/aqua/model/inline/ArrowInliner.scala

```diff
@@ -376,8 +378,14 @@ object ArrowInliner extends Logging {
       arrowArgs = args.arrowArgs(previousArrowsState)
       // Update states and rename tags
       renamedArrows <- updateArrowsAndRenameArrowArgs(arrowArgs ++ absArrows, fn, absRenames)
+
       argsToDataShouldRename <- updateExportsAndRenameDataArgs(args.dataArgs ++ absVars, absRenames)
-      allShouldRename = argsToDataShouldRename ++ renamedArrows ++ absRenames
+
+      // rename variables that store arrows
+      _ <- Exports[S].renameVariables(renamedArrows)
+
+      allShouldRename = argsToDataShouldRename ++ (renamedArrows -- absArrows.keySet) ++ absRenames
```

Comment:
Could you leave a comment about why we should rename arrows from abilities in `Exports`, but not in function body?